internal/packager/init.go

```diff
@@ -198,7 +216,17 @@ func initFromComposeFile(name string, composeFile string) error {
 	if err == nil {
 		parseEnv(string(envRaw), settings)
 	}
-	keys, err := ExtractVariables(string(composeRaw))
+	compose := make(map[interface{}]interface{})
```

Comment:
I think we should *just* use `github.com/docker/cli/compose/schema` package here (https://github.com/docker/cli/blob/master/cli/compose/schema/schema.go#L58) `schema.Validate` would validate that we support the composefile version and also that the composefile is valid.